google_guest_agent/uefi/uefi.go

```diff
@@ -0,0 +1,57 @@
+// Package uefi provides utility functions to read UEFI variables.
+package uefi
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	defaultEFIVarsDir = "/sys/firmware/efi/efivars"
+)
+
+// VariableName represents UEFI variable name and GUID.
+// Format: {VariableName}-{VendorGUID}
+type VariableName struct {
+	Name string
+	GUID string
+}
+
+// Variable represents UEFI Variable and its contents.
+type Variable struct {
+	Name       VariableName
+	Attributes []byte
+	Content    []byte
+}
+
+// VariablePath returns a path for UEFI variable on disk.
+func VariablePath(v VariableName) string {
+	if v.Name == "testname" && v.GUID == "testguid" {
```

Comment:
What about adding a `RootDir` field to `Variable`, if the `RootDir` is "" we use the default one, otherwise we assume it(or even some constructor to configure it, or any other option), that way we configure the `Variable.RootDir` with `t.TempDir()` in the unit tests, and remove this logic from this function?

Reply:
Done